tests/Test.ADAL.NET.Unit/Test.ADAL.NET.Unit.csproj

```diff
@@ -59,6 +59,10 @@
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.Data" />
+    <Reference Include="System.IdentityModel" />
+    <Reference Include="System.IdentityModel.Tokens.Jwt, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```

Comment:
I realised that the test `ClientAssertionWithX509Test` was not actually validating that the request crafted by ADAL contained a valid JWT (i.e., what if the certificate was used incorrectly?).
I added a new property to the `MockHttpMessageHandler` to allow tests to get access to the raw request and perform any additional validation.
Specifically for `ClientAssertionWithX509Test` I'm inspecting the request body to extract the "client_assertion" and then am using `JwtTokenHandler.CanReadToken` and `JwtSecurityToken` (see file _TokenHandler.cs_) to validate relevant parts of the header and payload.
I _could_ parse the JWT (unbase64, split '.' etc...) but I thought it easier to use the `System.IdentityModel.Tokens.Jwt` package (and since this is only the test project it doesn't impact the actual ADAL library's dependencies).

I also realise that in ADAL we also have JWTHeader and JWTPayload types but didn't want to use these to validate in the test since they're what's being used to craft the JWT in the first place 😃.